### PR TITLE
AB#33843: Net6

### PR DESCRIPTION
**Pull request by blargly**
Upgraded project to Net 6 as a precursor to implementing identy on the new directory.

Replaced startup/program launch with modern startup free approach.

Keen to hear opinions on how program.cs should be organised.

### CODE REVIEW

File: src/Submissions/Api/Api.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (AndyRae) [inline src/Submissions/Api/Api.csproj:4]: Moving to 6 - might be an opportunity to use [ImplicitUsings](https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#implicitusings);
Would be interested to hear opinions on effectively hiding many of the `using System;` directives, but would note in new 6 apps that this is enabled by default.

File: src/Submissions/Api/Api.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
```
Comment (beforan) [inline src/Submissions/Api/Api.csproj:4]: need to make sure all other projects which target net5 are changed to net6 as well.

e.g. `Data`

File: src/Submissions/Api/Program.cs [hunk 3]
```diff
@@ -1,39 +1,370 @@
...
+options.UseNpgsql("Omop"));
+
+builder.Services.AddDbContext<BiobanksDbContext>(options =>
+    options.UseSqlServer(connectionString,
+    sqlServerOptions => sqlServerOptions.CommandTimeout(300000000)));
+builder.Services.AddDatabaseDeveloperPageExceptionFilter();
+
+//identity
+builder.Services.AddDefaultIdentity<IdentityUser>(options => options.SignIn.RequireConfirmedAccount = true)
+    .AddEntityFrameworkStores<BiobanksDbContext>();
```
Comment (beforan) [inline src/Submissions/Api/Program.cs:67]: Does `BiobanksDbContext` also need updating to inherit from `IdentityDbContext` for this to work?

(That is a desirable change at some point anyway)

File: src/Submissions/Api/Program.cs [hunk 4]
```diff
@@ -1,39 +1,370 @@
...
+builder.Services.AddDbContext<OmopDbContext>(options =>
+options.UseNpgsql("Omop"));
+
+builder.Services.AddDbContext<BiobanksDbContext>(options =>
+    options.UseSqlServer(connectionString,
+    sqlServerOptions => sqlServerOptions.CommandTimeout(300000000)));
+builder.Services.AddDatabaseDeveloperPageExceptionFilter();
+
+//identity
+builder.Services.AddDefaultIdentity<IdentityUser>(options => options.SignIn.RequireConfirmedAccount = true)
```
Comment (beforan) [inline src/Submissions/Api/Program.cs:66]: `IdentityUser` should be `ApplicationUser` or whatever the custom user class is called. I know it's been copied over to Submissions already.

File: src/Submissions/Api/Program.cs [hunk 5]
```diff
@@ -1,39 +1,370 @@
...
+    .AddEntityFrameworkStores<BiobanksDbContext>();
+
+builder.Services.AddRazorPages();
```
Comment (beforan) [inline src/Submissions/Api/Program.cs:69]: Don't think Submissions uses Razor Pages, so don't add them

File: src/Submissions/Api/Program.cs [hunk 6]
```diff
@@ -1,39 +1,370 @@
...
+
+                // Hangfire Server
+                .UseHangfireDashboard();
+
+app.UseRouting();
+
+app.UseAuthentication();
+app.UseAuthorization();
+
+app.MapRazorPages();
```
Comment (beforan) [inline src/Submissions/Api/Program.cs:368]: As above, don't believe this is needed.

File: src/Submissions/Api/appsettings.json [hunk 7]
```diff
@@ -18,11 +18,7 @@
     "EntitiesPerSubmission": 10000
   },
 
-  "Logging": {
-    "LogLevel": {
-      "Default": "Warning"
-    }
-  },
+"Microsoft.AspNetCore":  "Warning",
```
Comment (beforan) [inline src/Submissions/Api/appsettings.json:21]: This seems wrong... pretty sure it should still be inside a logging section?

**Review by beforan — CHANGES_REQUESTED** [review]

File: src/Submissions/Api/Api.csproj (hunk 1 above)
Comment (beforan) [reply src/Submissions/Api/Api.csproj:4]: I would suggest as a starter for 10 we do what the net6 templates do by default for this.

But yeah, agree could be useful going forwards

File: src/Submissions/Api/Program.cs (hunk 4 above)
Comment (beforan) [reply src/Submissions/Api/Program.cs:66]: I appreciate that the actual identity config is still to come in a later PR, but using the wrong class here at this stage might cause undesirable extra migrations to get the right Users table schema.

**Review by AndyRae — DISMISSED** [review]

**Review by AndyRae — APPROVED** [review]

**Review by MoizIqbal97 — APPROVED** [review]

**Pull request merged**